libs/designer/src/lib/ui/panel/templatePanel/templatePanel.tsx

```diff
@@ -59,10 +60,27 @@ export const TemplatePanel = ({ onCreateClick }: { onCreateClick: () => Promise<
     [currentPanelView, templateTitle, templateDescription, manifest?.details, intlText.CREATE_WORKFLOW, intlText.BY_MICROSOFT]
   );
   const onRenderFooterContent = useCallback(
-    () => (selectedTabProps?.footerContent ? <TemplatesPanelFooter {...selectedTabProps?.footerContent} onClose={dismissPanel} /> : null),
+    () =>
+      selectedTabProps?.footerContent ? (
+        <TemplatesPanelFooter
+          {...selectedTabProps?.footerContent}
+          secondaryButtonOnClick={selectedTabProps?.footerContent?.secondaryButtonOnClick ?? dismissPanel}
+        />
+      ) : null,
     [selectedTabProps, dismissPanel]
   );
 
+  useEffect(() => {
+    async function fetchWorkflowNames() {
+      const existingWorkflowNames = await TemplateService()?.getExistingWorkflowNames();
+
+      dispatch(setExistingWorkflowNames(existingWorkflowNames));
+    }
```

Comment:
Use a query cache which will call templateserfvice and that method can be used as useExistingWOrkflowNames which is a react hook so we do not have to await